crates/bevy_winit/src/lib.rs

```diff
@@ -772,13 +788,27 @@ fn handle_winit_event(
         Event::UserEvent(RequestRedraw) => {
             runner_state.redraw_requested = true;
         }
+        Event::LoopExiting => {
+            let exit = exit_happened.clone().unwrap_or_else(|| {
+                error!("App exited without raising a `AppExit`!");
+                AppExit::error()
```

Comment:
```suggestion
                // Even though an error occurred, we still want to return an `AppExit` because it is preferable to panicking.
                AppExit::error()
```